lib/explorer/series.ex

```diff
@@ -2797,6 +2797,155 @@ defmodule Explorer.Series do
     do: Shared.apply_impl(series, :is_nan)
 
   def is_nan(%Series{dtype: dtype}), do: dtype_error("is_nan/1", dtype, [:float])
+  # Strings
+
+  @doc """
+  Detects whether a string contains a substring or a regular expression pattern.
+
+  ## Examples
+
+      iex> s = Explorer.Series.from_list(["abc", "def", "bcd"])
+      iex> Explorer.Series.contains(s, "bc")
+      #Explorer.Series<
+        Polars[3]
+        boolean [true, false, true]
+      >
+
+      iex> s = Explorer.Series.from_list(["abc1", "def", "bcd"])
+      iex> Explorer.Series.contains(s, ~r/\\d/)
+      #Explorer.Series<
+        Polars[3]
+        boolean [true, false, false]
+      >
+  """
+  @doc type: :element_wise
+  @spec contains(Series.t(), String.t() | Regex.t()) :: Series.t()
+  def contains(%Series{dtype: :string} = series, pattern)
+      when K.or(K.is_binary(pattern), K.is_struct(pattern, Regex)),
+      do: Shared.apply_impl(series, :contains, [pattern])
+
+  def contains(%Series{dtype: :string} = series, literal) when is_binary(literal),
+    do: Shared.apply_impl(series, :contains_literal, [literal])
```

Comment:
This is polars specific and I Don't think we should expose it here. :) In fact, given that you match on K.is_binary in the previous clause, I think this whole clause and all mentions to `contains_literal` can be removed?

Reply:
Ah I thought I had cleaned that up already! Must have snuck past me when I rebased.